src/service/db/organization.rs

```diff
@@ -125,39 +129,111 @@ pub async fn watch() -> Result<(), anyhow::Error> {
     }
 }
 
-pub async fn set(org: &Organization) -> Result<(), anyhow::Error> {
-    let key = format!("{ORG_KEY_PREFIX}/{}", org.identifier);
-    match db::put(
-        &key,
-        json::to_vec(org).unwrap().into(),
-        db::NEED_WATCH,
-        None,
+/// Cache the existing orgs in the beginning
+pub async fn cache() -> Result<(), anyhow::Error> {
+    let orgs = list(None).await?;
+    for org in orgs {
+        ORGANIZATIONS
+            .clone()
+            .write()
+            .await
+            .insert(org.identifier.clone(), org);
+    }
+    log::info!("Organizations Cached");
+    Ok(())
+}
+
+pub async fn watch() -> Result<(), anyhow::Error> {
+    let key = ORG_KEY_PREFIX;
+    let cluster_coordinator = db::get_coordinator().await;
+    let mut events = cluster_coordinator.watch(key).await?;
+    let events = Arc::get_mut(&mut events).unwrap();
+    log::info!("Start watching organizations");
+    loop {
+        let ev = match events.recv().await {
+            Some(ev) => ev,
+            None => {
+                log::error!("watch_orgs: event channel closed");
+                return Ok(());
+            }
+        };
+
+        if let db::Event::Put(ev) = ev {
+            let item_key = ev.key.strip_prefix(key).unwrap();
+            let item_value = ev.value.unwrap();
+            let json_val: Organization = if config::get_config().common.meta_store_external {
+                match get_org(item_key).await {
+                    Ok(val) => val,
+                    Err(e) => {
+                        log::error!("Error getting value: {}", e);
+                        continue;
+                    }
+                }
+            } else {
+                json::from_slice(&item_value).unwrap()
+            };
+            ORGANIZATIONS
+                .clone()
+                .write()
+                .await
+                .insert(item_key.to_string(), json_val);
+        }
+    }
+}
+
+pub async fn save_org(entry: &Organization) -> Result<(), anyhow::Error> {
+    let org_name = entry.name.trim();
+    if org_name.is_empty() {
+        return Err(anyhow::anyhow!("Organization name cannot be empty"));
+    }
+    if let Err(e) = organizations::add(
+        &entry.identifier,
+        org_name,
+        entry.org_type.as_str().parse().unwrap(),
```

Comment:
maybe it's worth to check if the `entry.org_type` is empty as well, since it's derived as `#[serde(default)]`, which based on below impl, it would be deserialized to `OrganizationType::Custom`. or that's as expected? 

```rs
impl FromStr for OrganizationType {
    type Err = ();

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        match s {
            "default" => Ok(OrganizationType::Default),
            "custom" => Ok(OrganizationType::Custom),
            _ => Ok(OrganizationType::Custom),
        }
    }
}
```

Reply:
Actually, this check is done in the `check_and_create_org()` and `check_and_create_org_without_ofga()` which are the only ways to create a `DEFAULT` org currently. Org created in any other way should be a custom org.